### ISSUE TITLE
Wrong transpose in src/align/detect_face.py

### ISSUE THREAD

**Issue by wuyongzheng** [davidsandberg/facenet#834]
In src/align/detect_face.py, np.transpose() is used to transpose input images before feeding to PNet, RNet and ONet. For example, the following code does transpose before ONet:

    tempimg[:,:,:,k] = imresample(tmp, (48, 48))
    ...
    tempimg1 = np.transpose(tempimg, (3,1,0,2))
    out = onet(tempimg1)

I think the correct one should be `np.transpose(tempimg, (3,0,1,2))`. The axis of `tempimg` is (height, width, color channel, batch). The expected axis of `onet`'s input is (batch, height, width, channels), as documented by tf.nn.conv2d.

The wrong code still works because it's a square image, i.e. height = width, and the training and detection are both wrong.

It's written this way, is it because the model is directly converted from the Matlab project without training? Matlab's convention is width x height x color.

**Comment by FortiLeiZhang** [comment]
Yes. The author just loaded the parameters from the MATLAB version without training, so the tensorflow version must be an exact translation from the MATLAB version. As there is a lot of .T in the MATLAB version, so there are a lot of transpose() in the TF version. This is also what I TuCao a lot when I was implementing my Keras version.

For more reference, please go to this [issue](https://github.com/ipazc/mtcnn/issues/4)

**Comment by wuyongzheng** [comment]
That's what I thought. Thanks.

**Comment by wuyongzheng** [comment]
That's what I thought. Thanks.
